src/Extensions/Diagnostics.HealthChecks/Internal/ServiceFabricHealthCheckPublisher.cs

```diff
@@ -2,43 +2,35 @@
 // Licensed under the MIT license.
 
 using System;
-using System.Collections.Generic;
 using System.Fabric;
-using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Extensions.Diagnostics.HealthChecks;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.ObjectPool;
 using Microsoft.Omex.Extensions.Abstractions;
 using SfHealthInformation = System.Fabric.Health.HealthInformation;
-using SfHealthState = System.Fabric.Health.HealthState;
 
 namespace Microsoft.Omex.Extensions.Diagnostics.HealthChecks
 {
-	internal class ServiceFabricHealthCheckPublisher : IHealthCheckPublisher
+	internal class ServiceFabricHealthCheckPublisher : HealthCheckPublisher
 	{
-		internal const string HealthReportSourceId = nameof(ServiceFabricHealthCheckPublisher);
-
-		internal const string HealthReportSummaryProperty = "HealthReportSummary";
-
 		private readonly IAccessor<IServicePartition> m_partitionAccessor;
 
 		private readonly ILogger<ServiceFabricHealthCheckPublisher> m_logger;
 
-		private readonly ObjectPool<StringBuilder> m_stringBuilderPool;
-
 		public ServiceFabricHealthCheckPublisher(
 			IAccessor<IServicePartition> partitionAccessor,
 			ILogger<ServiceFabricHealthCheckPublisher> logger,
-			ObjectPoolProvider objectPoolProvider)
+			ObjectPoolProvider objectPoolProvider) : base(objectPoolProvider)
 		{
 			m_partitionAccessor = partitionAccessor;
 			m_logger = logger;
-			m_stringBuilderPool = objectPoolProvider.CreateStringBuilderPool();
 		}
 
-		public Task PublishAsync(HealthReport report, CancellationToken cancellationToken)
+		internal static new string HealthReportSourceId => nameof(ServiceFabricHealthCheckPublisher);
```

Comment:
As a note for a future fix, If I understand everything correctly we should not use `new` here, instead we need to mark property in the base class as `virtual` and use `override` here.

Reply:
oh sorry, I missed static here, then I think it would be better to just make it not static.